fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java

```diff
@@ -171,31 +177,25 @@ public void savePersistentData() throws GeneralSecurityException {
   }
 
   /**
-   * launchLao is invoked when the user tries to create a new LAO. The method creates a `CreateLAO`
+   * createLao is invoked when the user tries to create a new LAO. The method creates a `CreateLAO`
    * message and publishes it to the root channel. It observers the response in the background and
    * switches to the home screen on success.
-   *
-   * @return a single containing the id of the launched lao if it was successful
    */
-  public Single<String> launchLao(String laoName) {
+  public void createLao(String laoName, String serverAddress) {
     Log.d(TAG, "creating lao with name " + laoName);
-    CreateLao createLao = new CreateLao(laoName, keyManager.getMainPublicKey());
-    Lao lao = new Lao(createLao.getId());
-
-    return networkManager
-        .getMessageSender()
-        .publish(keyManager.getMainKeyPair(), Channel.ROOT, createLao)
-        .doOnComplete(() -> Log.d(TAG, "got success result for create lao with id " + lao.getId()))
-        .andThen(networkManager.getMessageSender().subscribe(lao.getChannel()))
-        .toSingleDefault(lao.getId());
+    networkManager.connect(serverAddress);
+    getApplication()
+        .startActivity(
+            ConnectingActivity.newIntentForCreatingDetail(
+                getApplication().getApplicationContext(), laoName));
```

Comment:
Ideally, I would prefer to keep the structure where the view model does not change the fragment.

To me, it is the caller's job to do that.

I also join Jean on the Javadoc/method name that doesn't seem to be up-to-date with the actual behavior.

Reply:
Agreed, especially considering that the whole thing of going through the viewmodel was not relevant anymore imo